python/graphstorm/model/gnn_distill/student.py

```diff
@@ -0,0 +1,145 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    GNN student model class
+"""
+import os
+import logging
+import torch as th
+from torch import nn
+from transformers import (
+    AutoTokenizer,
+    AutoConfig,
+    AutoModel,
+    DistilBertConfig, 
+    DistilBertModel, 
+)
+
+SUPPORTED_MODEL = {
+"DistilBertModel": (DistilBertConfig, DistilBertModel),
+}
+
+PRETRAINED_MODEL = [
+"distilbert-base-uncased",
+"distilbert-base-cased",
+]
+
+class GSDistilledModel(nn.Module):
+    """ GraphStorm GNN distilled model class.
+    User specified Transformer-based model 
+    and a projection matrix will be initialized.
+
+    Parameters
+    ----------
+    lm_type : str
+        Model type for Transformer-based student model.
+    pre_trained_name : str
+        Name of pre-trained model.
+    checkpoint_path : str
+        Path of model checkpoint.
+    """
+    def __init__(self, lm_type=None, pre_trained_name=None, checkpoint_path=None):
+        super(GSDistilledModel, self).__init__()
+
+        if not checkpoint_path:
+            assert lm_type is not None, \
+                "HuggingFace Name of model architecture needs to be specified"
+            assert pre_trained_name is not None, \
+                "HuggingFace Name of pre-trained weights needs to be specified"
+            # TODO (HZ): need to test other HF models.
+            if lm_type not in SUPPORTED_MODEL:
+                raise ValueError(f'Model class {lm_type} is not supported.')
+            if pre_trained_name not in PRETRAINED_MODEL:
+                raise ValueError(f'Pre-trained model {pre_trained_name} is not supported.')
+
+            # initiate Transformer-based model
+            self.lm_config = SUPPORTED_MODEL[lm_type][0]()
+            self.lm = SUPPORTED_MODEL[lm_type][1](self.lm_config)
+
+            # load pre-trained tokenizers
+            self.tokenizer = AutoTokenizer.from_pretrained(pre_trained_name)
+
+            # load pre-trained parameters
+            self.lm = self.lm.from_pretrained(pre_trained_name)
+        else:
+            self.load_from_gs_checkpoint(checkpoint_path)
+
+        # TODO (HZ): support more distance-based loss 
+        self.loss = nn.MSELoss()
+
+    def load_from_gs_checkpoint(self, checkpoint_path):
+        """ Load student moddel from checkpoint_path.
```

Comment:
i thought we agreed to move save_model to the model class.

Reply:
I did another check and found for all the trainers, they define the save model function in trainer class but load model from model class. I just followed the convention.

Reply:
addressed by new commit.